hls4ml/templates/vivado/nnet_utils/nnet_graph.h

```diff
@@ -0,0 +1,582 @@
+#ifndef NNET_GRAPH_H_
+#define NNET_GRAPH_H_
+
+#include "nnet_common.h"
+#include "nnet_merge.h"
+#include "nnet_dense.h"
+#include "nnet_dense_resource.h"
+#include "nnet_activation.h"
+#include "nnet_array.h"
+#include "hls_stream.h"
+#include <string>
+#include <sstream>
+#include <math.h>
+
+namespace nnet {
+  enum flow {source_to_target=0, target_to_source=1};
+  enum aggr {aggr_sum=0, aggr_mean=1, aggr_max=2};
+  
+  struct graph_config
+  {
+    // Internal data type definitions
+    typedef float bias_t;
+    typedef float weight_t;
+    typedef float table_t;
+    
+    // Layer Sizes
+    static const unsigned n_node = 10;
+    static const unsigned n_edge = 20;
+    static const unsigned n_features = 3;
+    static const unsigned e_features = 4;
+    static const unsigned n_out = 4;
+    static const unsigned n_layers = 3;
+
+    // message-passing parameters
+    static const unsigned aggr = aggr_sum;
+    static const unsigned flow = source_to_target;
+    
+    // Resource reuse info
+    static const unsigned io_type = io_parallel;
+    static const unsigned reuse_factor = 1;
+    static const unsigned n_zeros = 0;
+
+    static const bool activate_final = false;

```

Comment:
should this really be a boolean? I saw `activate_final` set to `sigmoid` in other places. Is this set anywhere?

Reply:
This is from Vesal's original code. I think you're thinking of "final_act" but that is only in the front-end afaik... you're right though, I should probably update these naming conventions